### PR TITLE
Fix the hook that restarts the amavis server and causes it to detach

### PR DESCRIPTION
**Pull request by johansmitsnl**
See #686
This should fix the amavis issue.

### CODE REVIEW

**Comment by johansmitsnl** [comment]
@tomav a rebuild for this to please

File: Dockerfile [hunk 1]
```diff
@@ -115,8 +115,9 @@ RUN sed -i -e 's/include_try \/usr\/share\/dovecot\/protocols\.d/include_try \/e
 COPY target/dovecot/dovecot-ldap.conf.ext /etc/dovecot
 COPY target/postfix/ldap-users.cf target/postfix/ldap-groups.cf target/postfix/ldap-aliases.cf /etc/postfix/
 
-# Enables Spamassassin CRON updates
-RUN sed -i -r 's/^(CRON)=0/\1=1/g' /etc/default/spamassassin
+# Enables Spamassassin CRON updates and update hook for supervisor
+RUN sed -i -r 's/^(CRON)=0/\1=1/g' /etc/default/spamassassin && \
+    sed -i -r 's/^\$INIT restart/\supervisorctl restart amavis/g' /etc/spamassassin/sa-update-hooks.d/amavisd-new
```
Comment (gutmensch) [inline Dockerfile]: Should there really be a backslash in front of s?! No need to escape anything?

**Comment by gutmensch** [comment]
Thanks for looking into this and fixing it, @johansmitsnl!

File: Dockerfile (hunk 1 above)
Comment (johansmitsnl) [inline Dockerfile]: Your correct, fixed it

**Comment by johansmitsnl** [comment]
the failures are not related to the chance. They fail every now and then..

**Comment by tomav** [comment]
Merged

**Comment by init-js** [comment]
I've noticed that the process hierarchy of  amavis is different when it is started with init.d (`start-stop-daemon --start --quiet --pidfile /var/run/amavis/amavisd.pid --startas /usr/sbin/amavisd-new --`) versus when it is started with supervisorctl (`/usr/sbin/amavisd-new foreground`). I don't know amavis enough to know if this has subtle implications.

```
root@mx1:/# /etc/init.d/amavis stop; supervisorctl stop amavis; : make it is not running
Stopping amavisd: (not running).
amavis: ERROR (not running)
root@mx1:/# /etc/init.d/amavis start; sleep 5;
Starting amavisd: amavisd-new.
root@mx1:/# ps -eaf | grep amavisd
amavis   27567     0  4 22:28 ?        00:00:00 /usr/sbin/amavisd-new (master)
amavis   27573 27567  0 22:28 ?        00:00:00 /usr/sbin/amavisd-new (virgin child)
root     27587 26926  0 22:28 ?        00:00:00 grep --color=auto amavisd
root@mx1:/# /etc/init.d/amavis stop
Stopping amavisd: amavisd-new.
root@mx1:/# supervisorctl start amavis; sleep 5;
amavis: started
root@mx1:/# ps -eaf | grep amavisd
amavis   27696     5 11 22:30 ?        00:00:01 /usr/sbin/amavisd-new (master)
root     27716 26926  0 22:30 ?        00:00:00 grep --color=auto amavisd

```
In either case, it looks like it's the `master` process which is listening on the unix socket (`127.0.0.1:10024`)

**Pull request merged**